code/modules/supply/supply_console.dm

```diff
@@ -0,0 +1,499 @@
+
+/obj/machinery/computer/supplycomp
+	name = "Supply Shuttle Console"
+	desc = "Used to order supplies."
+	icon_screen = "supply"
+	req_access = list(ACCESS_CARGO)
+	circuit = /obj/item/circuitboard/supplycomp
+
+	/// Is this a public console (Confirm + Shuttle controls are not visible)
+	var/is_public = FALSE
+	/// Time of last request
+	var/reqtime = 0
+	/// Can we order special supplies
+	var/hacked = FALSE
+	/// Can we order contraband
+	var/can_order_contraband = FALSE
+	///the economy database this machine is connected to
+	var/datum/money_account_database/main_station/account_database
+	///the department account tethered to this supply console
+	var/datum/money_account/cargo_account
+
+/obj/machinery/computer/supplycomp/Initialize(mapload)
+	. = ..()
+	reconnect_database()
+
+/obj/machinery/computer/supplycomp/proc/reconnect_database()
+	if(is_station_level(z))
+		account_database = GLOB.station_money_database
+		cargo_account = account_database.get_account_by_department(DEPARTMENT_SUPPLY)
+	else
+		account_database = null
```

Comment:
Currently, a supply computer build on lavaland would constantly runtime upon using. You'd prob want to build in checks for this instead.

Reply:
fixed in next commit